Sources/Ink/Internal/Reader.swift

```diff
@@ -70,15 +70,17 @@ extension Reader {
     }
 
     @discardableResult
-    mutating func readCharacters(matching keyPath: KeyPath<Character, Bool>) throws -> Substring {
+    mutating func readCharacters(matching keyPath: KeyPath<Character, Bool>, limit: Int = Int.max) throws -> Substring {
         let startIndex = currentIndex
-
-        while !didReachEnd {
+        var readCount = 0
+        
+        while !didReachEnd && readCount < limit {
```

Comment:
You don't want to throw an error in case the limit was reached? With a wording like `limit`, that's what I'd expect from the API. If that's not intended, then perhaps use a name like `maxCount` or something would make it sound less like a failure condition? (just a thought)

Reply:
My intention wasn't for reaching the limit to be a failing state, so if you think maxCount would be clearer I'm happy to rename it.